src/modules/site-v2/base/views/tools/variant_annotation.py

```diff
@@ -74,37 +76,57 @@ def variant_annotation():
 
 
 
-@variant_annotation_bp.route('/query/interval', methods=['POST'])
+@variant_annotation_bp.route('/query/interval',                       methods=['POST'])
+@variant_annotation_bp.route('/query/interval/<string:species_name>', methods=['POST'])
 @cache.memoize(60*60)
-def query_interval():
+def query_interval(species_name=None):
 
   # Extract the query
   payload = json.loads(request.data)
   query = payload.get('query')
 
+  # Get the species from the URL, allowing undefined
+  if species_name:
+    try:
+      species = Species.get(species_name.replace('-', '_'))
+    except NotFoundError:
+      return abort(404)
+  else:
+    species = None
+
   # If query is valid, run it and return the results
   is_valid = verify_interval_query(query=query)
   if is_valid:
-    data = StrainAnnotatedVariant.run_interval_query(query=query)
+    data = StrainAnnotatedVariant.run_interval_query(query=query, species=species)
     return jsonify(data)
 
   # Otherwise, return an empty response
   return jsonify({})
 
 
 
-@variant_annotation_bp.route('/query/position', methods=['POST'])
+@variant_annotation_bp.route('/query/position',                       methods=['POST'])
+@variant_annotation_bp.route('/query/position/<string:species_name>', methods=['POST'])
 @cache.memoize(60*60)
-def query_position():
+def query_position(species_name=None):
 
   # Extract the query
   payload = json.loads(request.data)
   query = payload.get('query')
 
+  # Get the species from the URL, allowing undefined
+  if species_name:
+    try:
+      species = Species.get(species_name.replace('-', '_'))
```

Comment:
see if there is a better interface for getting the species from the URL that can be reused across all endpoints, eg: `Species.get_from_url(...)`.

Reply:
Fixed in https://github.com/AndersenLab/CAENDR/pull/373/commits/2a427db1832c241399ea008eaea9c3fb52278d9b, with minor tweaks in following two commits